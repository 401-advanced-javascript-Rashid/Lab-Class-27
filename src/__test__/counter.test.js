import React from 'react';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';
import Counter from '../app.js';


describe('<Counter />', () => {

  it('Assert that state is being transferred to the DOM', () => {
    let test = mount(<Counter />);
    test.find('#rss1').simulate('click');
    expect(test.find('#text').text()).toBe('Counter[-1]');
  });

  it('Assert DOM stability via snapshot', ()=> {
    expect(renderer.create(<Counter />)).toMatchSnapshot();
  });
});
