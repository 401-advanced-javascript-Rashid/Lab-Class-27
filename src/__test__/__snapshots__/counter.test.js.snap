// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> Assert DOM stability via snapshot 1`] = `
<section>
  <div>
    <div
      id="text"
    >
      <h2
        id="h2"
      >
        Cou
        <span
          id="offset"
        >
          nter[
          0
          ]
        </span>
      </h2>
    </div>
  </div>
  <button
    id="rss2"
    onClick={[Function]}
  >
    +
  </button>
  <button
    id="rss1"
    onClick={[Function]}
  >
    -
  </button>
</section>
`;
